services/core/shared/core/compat/sdk_v5/blockchainIndex.js

```diff
@@ -244,14 +250,17 @@ const updateBlockIndex = async job => {
 };
 
 const deleteIndexedBlocks = async job => {
```

Comment:
As the given method only delete a single block, in my opinion, it would be better to rename this method to `deleteIndexedBlock` and update the places where it has been called.

Reply:
No, this method is capable of deleting multiple blocks from the index and is used for the same within the `verifyAndIndexBlock` method a fork is detected. Please check line number 336:

```
		// Fork handling
		if (blockInfo && blockInfo.id !== block.id) {
			const [highestIndexedBlock] = await blocksDB.find({ sort: 'height:desc', limit: 1 }, ['height']);
			const blocksToRemove = await blocksDB.find(
				{
					propBetweens: [{
						property: 'height',
						from: block.height + 1,
						to: highestIndexedBlock.height,
					}],
					sort: 'height:desc',
					limit: highestIndexedBlock.height - block.height,
				},
				['id', 'generatorPublicKey'],
			);
			await deleteIndexedBlocksQueue.add({ blocks: blocksToRemove });
		}
```